packages/tracing-operations/src/services/operationsService.ts

```diff
@@ -58,10 +67,42 @@ export function operationsServiceBuilder(dbService: DBService) {
         errors: tracing.errors,
       };
     },
-    async recoverTracing(): Promise<ApiRecoverTracingResponse> {
-      genericLogger.info(`Recover tracing`);
-      await dbService.recoverTracing();
-      return Promise.resolve({});
+
+    async recoverTracing(
+      params: ApiRecoverTracingParams,
+      logger: Logger,
+    ): Promise<ApiRecoverTracingResponse> {
+      logger.info(`Recover data for tracingId: ${params.tracingId}`);
+
+      const tracing = await dbService.findTracingById(params.tracingId);
+      if (!tracing) {
+        throw tracingNotFound(params.tracingId);
+      }
+
+      if (
+        tracing.state === tracingState.completed ||
+        tracing.state === tracingState.pending
```

Comment:
let's list values that should be allowed.
This way possible new states do not pass the check automatically
```suggestion
        tracing.state !== tracingState.missing &&
        tracing.state !== tracingState.error
```